### PR TITLE
Add CMake configuration

### PR DESCRIPTION
**Pull request by aacostadiaz**
This PR adds the CMake configuration to build with SYCL using DPCPP. It also adds the `DPCPP_SYCL_TARGET` and `DPCPP_SYCL_ARCH` options to select different device targets.

### CODE REVIEW

File: examples/cute/tutorial/CMakeLists.txt [hunk 1]
```diff
@@ -43,23 +43,23 @@ else ()
...
 
-cutlass_example_add_executable(
-  tiled_copy
-  tiled_copy.cu
-)
+  cutlass_example_add_executable(
+    tiled_copy
+    tiled_copy.cu
+  )
 endif()
```
Comment (mehdi-goli) [inline examples/cute/tutorial/CMakeLists.txt:65]:
```suggestion
endif()

```

**Review by mehdi-goli — APPROVED** [review]
LGTM! Thanks for the fix

**Review by AD2605 — APPROVED** [review]

**Comment by AD2605** [comment]
LGTM

File: CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +59,10 @@ project(CUTLASS VERSION ${_CUTLASS_VERSION_MAJOR}.${_CUTLASS_VERSION_MINOR}.${_C
 
 ################################################################################
 
+if (CUTLASS_ENABLE_SYCL)
+  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --allow-unsupported-compiler")
```
Comment (rolandschulz) [inline CMakeLists.txt:63]: What's the motivation for this? How are CMAKE_CUDA_FLAGS used with CUTLASS_ENABLE_SYCL?

File: CMakeLists.txt [hunk 3]
```diff
@@ -327,10 +331,11 @@ endif()
...
 if(CUTLASS_ENABLE_SYCL)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -fsycl -ffast-math -mllvm -enable-global-offset=false -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --cuda-gpu-arch=sm_${CUTLASS_NVCC_ARCHS}")
-  set(CMAKE_CUDA_ARCHITECTURES "${CUTLASS_NVCC_ARCHS}")
-  add_definitions( -DCUTLASS_ENABLE_SYCL=${CUTLASS_ENABLE_SYCL} )
+  set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+  add_compile_definitions(CUTLASS_ENABLE_SYCL)
+  set(DPCPP_SYCL_TARGET "nvptx64-nvidia-cuda" CACHE STRING "Default target for Nvidia GPU")
+  find_package(DPCPP)
```
Comment (rolandschulz) [inline CMakeLists.txt:338]: Why use this approach?
Isn't the BKM to use find_package(IntelSYCL REQUIRED) (https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2024-1/use-cmake-with-the-compiler.html)

**Review by rolandschulz — COMMENTED** [review]
Sorry didn't see this before.

File: README-sycl.md [hunk 4]
```diff
@@ -54,7 +52,7 @@ ninja [EXAMPLE_NAME]_sycl
 You can run it like this from your build directory
 
 ```
-LD_LIBRARY_PATH=/path/to/sycl/install/lib ./examples/cute/tutorial/[EXAMPLE_NAME]_sycl
+./examples/cute/tutorial/[EXAMPLE_NAME]_sycl
```
Comment (rolandschulz) [inline README-sycl.md:55]: why is LD_LIBRARY_PATH removed here? And then added back in #22?

**Pull request merged**